src/libpmem2/badblocks_ndctl.c

```diff
@@ -263,7 +263,8 @@ pmem2_badblock_context_new(const struct pmem2_source *src,
 
 	*bbctx = NULL;
 
-	if (ndctl_new(&ctx)) {
+	errno = ndctl_new(&ctx);
```

Comment:
this is wrong - ndctl_new doesn't return errno, it returns -errno.

Reply:
Done